sdk/iOS/test/MSCoreDataStoreTests.m

```diff
@@ -47,22 +50,72 @@ -(void)testInit
     XCTAssertNotNil(self.store, @"store creation failed");
 }
 
+// Tests MSCoreDataStore Read, Update and Delete operations are case insensitive for the Id column.
+-(void)testReadUpdateDelete_CaseInsensitiveIdOperations
+{
+    NSError *error;
+
+    // Seed data to be inserted in the table
+    NSString * const originalItemId = @"itemid";
+    NSString * const originalText = @"original text";
+    NSArray * const originalItems = @[ @{ @"id":originalItemId, @"text":originalText } ];
+
+    // Data to be used for updating the table. The Id differs from the Id of the seed data only in case
+    NSString * const updatedItemId = @"ITEMID";
+    NSString * const updatedText = @"updated text";
+    NSArray * const updatedItems = @[ @{ @"id":updatedItemId, @"text":updatedText } ];
```

Comment:
nit style: @"id" : updatedItemId (always use a space to either side of the : in the definition)